build.zig

```diff
@@ -114,8 +126,11 @@ fn compileRaylib(b: *std.Build, target: std.Build.ResolvedTarget, optimize: std.
         // Note: Currently using a suboptimal `O(m*n)` time algorithm where:
         // `m` corresponds roughly to the number of lines in `src/config.h`
         // `n` corresponds to the number of user-specified flags
-        outer: for (config_h_flags) |flag| {
+        outer: for (config_h_flags) |flag_val| {
+            const flag = flag_val[0];
+
             // If a user already specified the flag, skip it
+            config_iter.reset();
```

Comment:
This line was supposed to be here since #4398, but I somehow forgot to include it. Without it, one can pass custom flags but won't be able to properly override existing ones: Both the default and overriding flags will end up being passed.